internal/constants/scripts.go

```diff
@@ -66,140 +70,149 @@ function map_host_macs_to_interfaces() {
   for nic in $( ls $SYS_CLASS_NET_DIR )
   do
     mac=$(cat $SYS_CLASS_NET_DIR/$nic/address | tr '[:lower:]' '[:upper:]')
-    host_macs_to_hw_iface[$mac]=$nic
+    [[ -n "$mac" ]] && host_macs_to_hw_iface[$mac]=$nic
   done
 }
 
-function find_host_directory_by_mac_address() {
-  for d in $(ls -d ${NMCONNECTIONS_DIR}/host*)
-  do
+# Copy all nmstate files representing a host nic
+function copy_physical_nics_files() {
+  host_src_dir=$1
+  mapping_file=$2
+  host_work_dir=$3
+  for mac_address in $(cat $mapping_file | cut -d= -f1 | tr '[:lower:]' '[:upper:]');  do
+    host_nic="${host_macs_to_hw_iface[${mac_address}]:-}"
+    if [[  -n "$host_nic" ]]; then
+      logical_nic="${logical_mac_nic_map[${mac_address}]:-}"
+      if [[ -n "$(ls -A $host_src_dir/*.nmconnection )" ]] ; then
+        for nmfile in $host_src_dir/*.nmconnection ; do
+          filename=$(basename $nmfile)
+          prefix="${filename%%.*}"
+          extension="${filename#*.}"
+          [[ -n "$prefix" ]] && [[ "$prefix" == "$logical_nic" ]] &&  cp $nmfile $host_work_dir/${host_nic}.${extension} && rm -f $nmfile
+        done
+      fi
+    fi
+  done
+}
 
-    mapping_file="${d}/${MAC_NIC_MAPPING_FILE}"
+# If any nmstate files of the current host directory were copied, copy the remaining nmstate files
+function copy_other_files() {
+  host_src_dir=$1
+  host_work_dir=$2
+  if [ "$(ls -A $host_work_dir )" ] && [[ -n "$(ls -A ${host_src_dir}/*.nmconnection)" ]] ; then
+    for nmfile in ${host_src_dir}/*.nmconnection ; do
+      filename=$(basename $nmfile)
+      target_file=$host_work_dir/$filename
+
+      # If the same filename was not used already, then use it. Otherwise generate another name.
+      if ! [[ ${allocated_nics[$filename]} ]] ; then
```

Comment:
Only these "other" files make it into `allocated_nics`, right?
That means that we could overwrite a "real" file with a duplicated one here I think.

Do you think it would be safer/easier to fail this copy instead of trying to recover with a generated name?

Reply:
The files that represent hw nics are renamed to their corresponding nics.  Assuming we have 2 `bond0` files in 2 nmstates.  The first one will use the name `bond0` and the second one will use an extension

Reply:
I still think this should be a failure case.

I don't think we should be creating a new name when the user already had the chance to pick a correct one.

Reply:
What should we do for failure case? skip it?

Reply:
How visible of an error can we create? Anything we print should be in the journal, right?

Reply:
I am not sure.
With this change we rename files that start with physical nic names.  This causes files having same local nic names to be renamed to different names which means they will coexist together.
Example: `eth0.nmconnection` in 2 different directories might be renamed to `ens2.nmconnection` and `ens3.nmconnection`.
If the interface is virtual (for example bond), then if we have file `bond0.nmconnection` in 2 different directories, this will fail while it is possible that it will succeed.  Why should we fail it if we allow similar duplication for physical nics ?